modules/tinymce/src/themes/silver/main/ts/ui/sizeinput/SizeInput.ts

```diff
@@ -8,20 +8,21 @@ import {
   Keying,
   NativeEvents, Representing, SketchSpec, Tabstopping, Tooltipping
 } from '@ephox/alloy';
-import { Id, Optional, Unicode } from '@ephox/katamari';
+import { Cell, Fun, Id, Optional, Optionals, Unicode } from '@ephox/katamari';
 
 import { formChangeEvent, formInputEvent } from 'tinymce/themes/silver/ui/general/FormEvents';
 
 import { UiFactoryBackstageProviders } from '../../backstage/Backstage';
 import * as UiState from '../../UiState';
+import { onControlAttached, onControlDetached } from '../controls/Controls';
 import * as Icons from '../icons/Icons';
 import { formatSize, makeRatioConverter, noSizeConversion, parseSize, SizeConversion } from '../sizeinput/SizeInputModel';
 
 interface RatioEvent extends CustomEvent {
   readonly isField1: boolean;
 }
 
-export interface SizeInputGenericSpec {
+export interface SizeInputGenericSpec<ApiType = never> {
```

Comment:
I see that this was a bit of a struggle since I made this a generic thing shared between dialog and the context forms and the dialogs doesn't have a API or setup so I guess `never` is the appropriate type to default to.